projects/client/RabbitMQ.Client/src/client/impl/AutorecoveringConnection.cs

```diff
@@ -64,15 +64,17 @@ internal sealed class AutorecoveringConnection : IConnection
 
         private readonly object _recordedEntitiesLock = new object();
 
-        private readonly List<AutorecoveringModel> _models = new List<AutorecoveringModel>();
+        private readonly IDictionary<string, RecordedExchange> _recordedExchanges = new Dictionary<string, RecordedExchange>();
 
-        private readonly ConcurrentDictionary<RecordedBinding, byte> _recordedBindings =
-            new ConcurrentDictionary<RecordedBinding, byte>();
```

Comment:
No need for `ConcurrentDictionary` as operations are locked on `_recordedEntitiesLock`